src/document.c

```diff
@@ -519,10 +559,11 @@ FIELD_PREPROCESSOR(tagPreprocessor) {
   if (fdata->tags == NULL) {
     return 0;
   }
-  if (FieldSpec_IsSortable(fs)) {
+  if (FieldSpec_IsSortable(fs) && (field->unionType == FLD_VAR_T_CSTR ||
+                                   field->unionType == FLD_VAR_T_RMS)) {
```

Comment:
So what if it is defined as sortable and the return value is an array? I see we will just not set the sorting table, we need a test for that (to make sure we are not crashing/leaking). Also, we need to define the behavior here, I find it very confusing that some documents will have sortable value and some are not .. @emmanuelkeller FYI

Reply:
I agree about the behavior. We can't keep the current behavior since it is only practical when we have a single string. @emmanuelkeller WDYT?

Reply:
@emmanuelkeller @rafie we need to talk about this ^^